management_api_app/api/routes/workspaces.py

```diff
@@ -22,7 +23,12 @@ async def retrieve_active_workspaces(workspace_repo: WorkspaceRepository = Depen
 
 
 @router.post("/workspaces", status_code=status.HTTP_202_ACCEPTED, response_model=WorkspaceIdInResponse, name=strings.API_CREATE_WORKSPACE)
-async def create_workspace(workspace_create: WorkspaceInCreate, workspace_repo: WorkspaceRepository = Depends(get_repository(WorkspaceRepository))) -> WorkspaceIdInResponse:
+async def create_workspace(workspace_create: WorkspaceInCreate,
+                           workspace_repo: WorkspaceRepository = Depends(get_repository(WorkspaceRepository)),
+                           workspace_template_repo: WorkspaceTemplateRepository = Depends(get_repository(WorkspaceTemplateRepository))) -> WorkspaceIdInResponse:
+
+    ValidateRequest.validate_request(workspace_create, workspace_template_repo)
```

Comment:
validation is done outside try/catch and also requires extra db calls - this would be better placed inside the create_workspace_item code so that the template can be reused.